submission_worker.py

```diff
@@ -0,0 +1,300 @@
+
+    # make sure that the challenge base directory exists
+    create_dir(BASE_CHALLENGE_DATA_DIR)
+
+    active_challenges = Challenge.objects.filter(**q_params)
+
+    for challenge in active_challenges:
+        phases = challenge.challengephase_set.all()
+        extract_challenge_data(challenge, phases)
+
+
+def extract_submission_data(submission_id):
+    '''
+        * Expects submission id and extracts input file for it.
+    '''
+
+    try:
+        submission = Submission.objects.get(id=submission_id)
+    except Submission.DoesNotExist:
+        print 'Submission {} does not exist'.format(submission_id)
+
+    submission_input_file = submission.input_file
+    submission_data_directory = SUBMISSION_DATA_DIR.format(submission_id=submission.id)
+    # create submission directory
+    create_dir(submission_data_directory)
+
+    submission_zip_file = join(submission_data_directory, 'submission_{}.zip'.format(submission.id))
+    download_and_extract_file(submission_input_file, submission_zip_file, submission_data_directory)
+
+
+def run_submission(challenge_id, phase_id, submission_id, user_annotation_file_path):
+    '''
+        * receives a challenge id, phase id and user annotation file path
+        * checks whether the corresponding evalutaion script for the challenge exists or not
+        * checks the above for annotation file
+        * calls evaluation script via subprocess passing annotation file and user_annotation_file_path as argument
+    '''
+    evaluation_script_path = CHALLENGE_DATA_DIR.format(challenge_id=challenge_id)
+    phase_data_dir = PHASE_DATA_DIR.format(challenge_id=challenge_id, phase_id=phase_id)
+    metadata_file_path = join(CHALLENGE_DATA_DIR.format(challenge_id=challenge_id), 'metadata')
+    submission_data_dir = SUBMISSION_DATA_DIR.format(submission_id=submission_id)
+    # hard coded, should be fetched from challenge model
+    execution_time_limit = 5 * 60
+    with open(metadata_file_path, 'r') as metadata_file:
+        prog_info = yaml.load(metadata_file.read())
+
+    prog_cmd_list = []
+    if 'command' in prog_info:
+        if isinstance(prog_info['command'], type([])):
+            prog_cmd_list = [_.strip() for _ in prog_info['command']]
+        else:
+            prog_cmd_list = [prog_info['command'].strip()]
+    if len(prog_cmd_list) <= 0:
+        raise Exception('Program command is not specified.')
+
+    stdout_file_name = 'temp_stdout.txt'
+    stderr_file_name = 'temp_stderr.txt'
+
+    try:
+        submission = Submission.objects.get(id=submission_id)
+    except Submission.DoesNotExist:
+        submission = None
+
+    stdout_file = join(submission_data_dir, stdout_file_name)
+    stderr_file = join(submission_data_dir, stderr_file_name)
+
+    stdout = open(stdout_file, 'a+')
+    stderr = open(stderr_file, 'a+')
+
+    for prog_cmd_counter, prog_cmd in enumerate(prog_cmd_list):
+        prog_cmd = prog_cmd.replace("$program", evaluation_script_path) \
```

Comment:
Program should be replaced by the executable call, right? It looks like this will just contain the path to the executable but not the actual executable. 

@deshraj : Are we capturing this information somewhere in the UI? Or are we assuming a standard name to the file. For ex: "python evaluate.py $input $output" ?